airbyte-integrations/connectors/source-metabase/source_metabase/source.py

```diff
@@ -102,17 +34,9 @@ def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]) ->
                 session.close_session()
 
     def streams(self, config: Mapping[str, Any]) -> List[Stream]:
-        self.session = MetabaseAuth(logging.getLogger("airbyte"), config)
-        if not self.session.has_valid_token():
-            raise ConnectionError("Failed to connect to source")
-        args = {"authenticator": self.session, API_URL: config[API_URL]}
-        return [
-            Activity(**args),
-            Cards(**args),
-            Collections(**args),
-            Dashboards(**args),
-            Users(**args),
-        ]
+        self.session = MetabaseAuth(config, {})
+
+        return super().streams(config)
```

Comment:
Once again: you're not passing session object here. This means new session will be instantiated per each stream if an authenticator is passed to a stream in yaml. After the sync succeeds, only the source session (defined two lines above) will be closed, but not those sessions in the streams, right?